CMakeLists.txt

```diff
@@ -1,16 +1,27 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.0)
 project(azmq)
 
 include(AzmqCPack.cmake)
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
+
+find_package(Boost 1.48 COMPONENTS system date_time thread chrono random REQUIRED)
+find_package(ZeroMQ 4.0 REQUIRED)
+
+find_package(Threads REQUIRED)
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+find_package(Threads REQUIRED)
+if (NOT CMAKE_THREAD_LIBS_INIT)
+    set(CMAKE_THREAD_LIBS_INIT "")
+endif ()
+
 set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LANGUAGE_STANDARD "c++11")
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
```

Comment:
this does not match for `Apple-Clang!`
```cmake
if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Reply:
yes it doesnt.

I don't like the practice of setting -Wall for the users of the library, but this was a layover from what was previously done.

For our dev build we can turn up the warnings level way up.. Perhaps we should even bring in some of the excellent templates from Jason Turnings starter project - This would fix the apple-clang issue

https://github.com/cpp-best-practices/cpp_starter_project/blob/main/cmake/CompilerWarnings.cmake

Reply:
Yes, a good `template project`, only installing is missing.
The cmake modules are clean, well designed, and good!